packages/dds/tree/src/core/edit-manager/editManager.ts

```diff
@@ -309,6 +316,26 @@ export class EditManager<
 		return this.changeFamily.intoDelta(change);
 	}
 
+	/**
+	 * Fast-forwards the local branch.
+	 * @param newHead - the new head of the local branch. This must be a descendant of the current local branch head.
+	 * @returns the composition of changes from the previous head to the new head of the local branch
+	 */
+	public fastForwardLocalBranch(newHead: GraphCommit<TChangeset>): TChangeset {
+		const changes: GraphCommit<TChangeset>[] = [];
+		assert(
```

Comment:
The pull happens in 'merge'. Since we are moving the pull inside of the mergeIntoBase method, it will have to happen here too now. I.e. every implementer of 'mergeIntoBase' has to call 'pull' rather than 'pull' being called by 'merge'. I guess that's the downside to moving it inside.